tests/perf-test/ReportScreen.perf-test.js

```diff
@@ -101,6 +101,34 @@ afterEach(() => {
     PusherHelper.teardown();
 });
 
+/**
+ * This is a helper function to create a mock for the addListener function of the react-navigation library.
+ * The reason we need this is because we need to trigger the transitionEnd event in our tests to simulate
+ * the transitionEnd event that is triggered when the screen transition animation is completed.
+ *
+ * P.S: This can't be moved to a utils file because Jest wants any external function to stay in the scope.
+ *
+ * @returns {Object} An object with two functions: triggerTransitionEnd and addListener
+ */
+const createAddListenerMock = () => {
+    const transitionEndListeners = [];
+    const triggerTransitionEnd = () => {
+        transitionEndListeners.forEach((transitionEndListener) => transitionEndListener());
+    };
+
+    const addListener = jest.fn().mockImplementation((listener, callback) => {
+        if (listener === 'transitionEnd') {
+            transitionEndListeners.push(callback);
+        }
+        return () => {
+            // eslint-disable-next-line rulesdir/prefer-underscore-method
+            transitionEndListeners.filter((cb) => cb !== callback);
```

Comment:
Why can't we use the underscore function?

Reply:
We can, did it. 👍